### DISCUSSION TITLE
How To Configure @cypress/grep plugin v3.1.0 for Cypress v10.11.0 (JS)

### DISCUSSION THREAD

**Discussion by seriousAndInteresting** [cypress-io/cypress#25182] [Questions and Help]
Thanks @bahmutov for creating the [@cypress/grep](https://github.com/cypress-io/cypress/tree/develop/npm/grep#cypressgrep) plugin. 
I have trouble with selecting tests
Here is where I've added this plugin
![image](https://user-images.githubusercontent.com/115154617/207926011-ff323097-de83-40a0-88e2-4ed15868551f.png)

Here is describe with tests which i run with the command  => npx cypress run --env grepTags=@smoke

![image](https://user-images.githubusercontent.com/115154617/207926327-1396c198-f20a-4ff4-ba23-5eb1aa3c08ae.png)

And my output in terminal
![image](https://user-images.githubusercontent.com/115154617/207926738-0608abbc-99df-4ac1-b4e9-d18b164fdf71.png)

Maybe I forgot some place to add?

**Comment by portyu9** [comment]
Hi @seriousAndInteresting 

The problem is in the support-file registration.

Your screenshot shows:

```js
require('@cypress/grep')
```

but with the version you were using, simply requiring the package is not enough. The registration function needs to be called.

For the `@cypress/grep` version from this Cypress 10/11 era, use:

```js
// cypress/support/e2e.js
const registerCypressGrep = require('@cypress/grep')

registerCypressGrep()
```

Then the command you were using can filter the tagged tests:

```bash
npx cypress run --env grepTags=@smoke
```

There has since been a breaking API change in newer `@cypress/grep` releases. With the **current** package, registration is:

```js
const { register: registerCypressGrep } = require('@cypress/grep')

registerCypressGrep()
```

and current Cypress / `@cypress/grep` versions use exposed configuration, for example:

```bash
npx cypress run --expose grepTags="@smoke"
```

So don't mix the current README's registration/CLI syntax with the old `3.1.0` package shown in the screenshots.

For your original setup, the missing `registerCypressGrep()` call is the first thing to fix.

Please mark this answer as accepted if it helped, thank you.
